firebase-sessions/src/main/kotlin/com/google/firebase/sessions/SessionFirelogPublisher.kt

```diff
@@ -71,8 +78,51 @@ internal class SessionFirelogPublisher(
     }
   }
 
-  /** Gets the Firebase Installation ID for the current app installation. */
-  private suspend fun getFid() =
+  /** Determines if the SDK should log a session to Firelog. */
+  private suspend fun shouldLogSession(
+    subscribers: Map<SessionSubscriber.Name, SessionSubscriber>
+  ): Boolean {
+    if (subscribers.isEmpty()) {
+      Log.d(
+        SessionLifecycleService.TAG,
+        "Sessions SDK did not have any dependent SDKs register as dependencies. Events will not be sent."
+      )
+      return false
+    }
+
+    if (subscribers.values.none { it.isDataCollectionEnabled }) {
```

Comment:
why would AQS data collection be disabled on a subscriber?

Reply:
We need at least one product sdk to have data collection enabled or we don't have consent to upload sessions